File: Task/Strip-comments-from-a-string/Nim/strip-comments-from-a-string.nim
import strutils

proc removeComments(line: string; sep: char): string =
  line.split(sep)[0].strip(leading = false)

const
  Str1 = "apples, pears # and bananas"
  Str2 = "apples, pears ; and bananas"

echo "Original: “$#”" % Str1
echo "Stripped: “$#”" % Str1.removeComments('#')
echo "Original: “$#”" % Str2
echo "Stripped: “$#”" % Str2.removeComments(';')

File: Task/Strip-comments-from-a-string/PascalABC.NET/strip-comments-from-a-string.pas
function RemoveComments(s,delim: string): string
  := Regex.Replace(s, delim + '.+', '');

begin
  Writeln(RemoveComments('apples, pears # and bananas','#'));
  Writeln(RemoveComments('apples, pears ; and bananas',';'));
end.
